pkg/addons/addons.go

```diff
@@ -443,6 +443,21 @@ func Start(wg *sync.WaitGroup, cc *config.ClusterConfig, toEnable map[string]boo
 			klog.Errorf("store failed: %v", err)
 		}
 	}
+
+	loadAddonsListToClusterConfig(cc)
+
+}
+
+// load addons list with status in to cluster configs on start
+func loadAddonsListToClusterConfig(cc *config.ClusterConfig) {
```

Comment:
this doesnt seem to be the right place to fix it, by the the time that addons are enabled, the config already has the addons.

Reply:
Yuh, that only happens when we enable a particular addon. Will check again and fix this

Reply:
So basically Addons filed in cluster config gets updated when we enable addons. Not in the start. But you can check that addons status by using `assets.Addons["auto_pause"].IsEnabled`. Would you mind explaining the missing point here?

Reply:
I checked by enabling the "auto-pause" addon and it's getting successfully added in tot the cluster configs. But if you need to load all the addons in to cluster-config (which is not necessary as we can achieve this through the above method I've mentioned), we'll have to do it on `minikube start` right?

Reply:
@BLasan the right place to fix this is in node package or start.go the idea is the ClusterConfig in the start.go was not filled correclty with addons, please feel free to close this PR if this seems to be a big refactor.

Reply:
This `addons.Start()` function gets called from `Start()` function inside the `pkg/minikube/node/start.go`. That `node.Start()` function gets called from `startWithDriver()` function inside the `cmd/minikube/cmd/start,go`. That function will be called by `runStart()` function which handles the execution flow of `minikube start`